daemon-scheduler/pkg/deployment/deployment.go

```diff
@@ -151,10 +151,11 @@ func (d deployment) CreateSubDeployment(ctx context.Context, environmentName str
 			"There is no deployment for environment with name '%s' to create a sub-deployment", environmentName)
 	}
 
-	return d.startSubDeployment(ctx, env, deployment, instanceARNs)
+	return d.startDeployment(ctx, env, deployment, instanceARNs)
```

Comment:
GetCurrentDeployment has not changed in this PR. In the next one getCurrentDeployment won't include pending anymore.